flixel/FlxG.hx

```diff
@@ -367,10 +367,10 @@ class FlxG
 	 * Attempts to switch from the current game state to `nextState`.
 	 * The state switch is successful if `switchTo()` of the current `state` returns `true`.
 	 */
-	public static inline function switchState(nextState:FlxState):Void
+	public static inline function switchState(nextState:() -> FlxState):Void
 	{
-		if (state.switchTo(nextState))
-			game._requestedState = nextState;
+		if (state.switchTo(nextState()))
```

Comment:
state.switchTo() should take a `()->FlxState` rather than a FlxState instance, it should store the function in `_requestedState` and  wait until the new stuff is all set up to call it, creating the state